File: Solutions/C#/Accepted/226-invert-binary-tree.cs
/*
Problem Title: 226 - Invert Binary Tree
Lang: C#
Runtime: 92 ms
Memory: 24 MB
Problem Url: https://leetcode.com/problems/invert-binary-tree
Submission Url: https://leetcode.com/problems/invert-binary-tree/submissions/305609487
*/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if (root == null){
            return root;
        }
        TreeNode temp = root.left;
        root.left = InvertTree(root.right);
        root.right = InvertTree(temp);
        return root;
    }
}

File: Solutions/C++/Accepted/226-invert-binary-tree.cpp
/*
Problem Title: 226 - Invert Binary Tree
Lang: C++
Runtime: 0 ms
Memory: 8.3 MB
Problem Url: https://leetcode.com/problems/invert-binary-tree
Submission Url: https://leetcode.com/problems/invert-binary-tree/submissions/347674325
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        
        if(root == NULL){
            return root;
        }
        
        TreeNode* temp = root->left;
        
        root->left = invertTree(root->right);
        root->right = invertTree(temp);
        
        return root;
    }
};